### PR TITLE
Allow non basic users to make instant updates without approval

### PR DESCRIPTION
**Pull request by danmitchell-**
screen shot 2017-10-06 at 15 20 37

### CODE REVIEW

File: spec/features/publish_record_spec.rb [hunk 1]
```diff
@@ -68,18 +66,8 @@ def stub(url, file_path)
     click_button 'Continue'
 
     expect(page).to have_content 'Check the details of your new country'
-    click_button 'Submit'
```
Comment (gidsg) [inline spec/features/publish_record_spec.rb]: Might be nice to test this approval flow in another scenario? And a basic user submitting a change to be reviewed.
Happy for this to be a separate PR though if it is non-trivial work.

File: app/controllers/register_controller.rb [hunk 2]
```diff
@@ -63,20 +63,50 @@ def create
...
+      redirect_to action: 'index', register: params[:register]
+    else
+      if params[:confirm_approve] == '1'
+        @change = Change.new(register_name: params[:register], payload: payload, user_id: current_user.id)
+        @change.status = Status.new(status: 'approved', reviewed_by_id: current_user.id)
+        @change.save
+
+        rsf_body = CreateRsf.(@change.payload, @change.register_name)
+
+        Rails.env.development? || Rails.env.staging? || response = RegisterPost.(@change.register_name, rsf_body)
```
Comment (gidsg) [inline app/controllers/register_controller.rb]: Since this is repeated could we move it to a variable e.g. `is_non_prod = Rails.env.development? || Rails.env.staging?`

File: app/controllers/register_controller.rb [hunk 3]
```diff
@@ -63,20 +63,50 @@ def create
 
     payload = generate_canonical_object(fields, params)
 
-    @change = Change.new(register_name: params[:register], payload: payload, user_id: current_user.id)
-    @change.status = Status.new(status: 'pending')
-    @change.save
+    if current_user.basic?
+      @change = Change.new(register_name: params[:register], payload: payload, user_id: current_user.id)
```
Comment (gidsg) [inline app/controllers/register_controller.rb]: Since this `Change.new` is the same for both user types. We could make it a lambda? e.g.
`create_change = lambda { Change.new(register_name: params[:register], payload: payload, user_id: current_user.id) }`

`@change = create_change.call`

File: app/controllers/register_controller.rb (hunk 2 above)
Comment (danmitchell-) [reply app/controllers/register_controller.rb]: I feel like this whole block is wrong anyway i.e. we should only run this if prod and not check for all the other environments

Comment (gidsg) [reply app/controllers/register_controller.rb]: yeah, possibly although I think the end to end tests in env `test` expect the POST to happen then mock the response, which is a useful part of the test. Maybe we could make this a per env configuration value like `post_to_register`?

Comment (danmitchell-) [reply app/controllers/register_controller.rb]: You are right, our tests would fail but I dont think thats a reason to not refactor... maybe another story :)

File: spec/features/publish_record_spec.rb (hunk 1 above)
Comment (danmitchell-) [reply spec/features/publish_record_spec.rb]: Yeah, this is a tough one so a separate PR would be a better idea

File: app/controllers/register_controller.rb (hunk 3 above)
Comment (danmitchell-) [reply app/controllers/register_controller.rb]: I don't we think we should be doing this at all here. We should be using proper resources i.e. `foo.com/country/AU/change/new` or `foo.com/country/new`

File: spec/features/publish_record_spec.rb (hunk 1 above)
Comment (danmitchell-) [reply spec/features/publish_record_spec.rb]: Agreed but non-trivial :)
